### PR TITLE
vtk-m: update to latest release

### PR DESCRIPTION
**Pull request by vicentebolea**
Finally! After a quiet spring, VTK-m v1.8.0 is finally out :tada: :tada: :tada:

This bring exiting new features, and bugfixes, find the tentative release notes at https://gitlab.kitware.com/vtk/vtk-m/-/blob/master/docs/changelog/1.8/release-notes.md

Let me know if we need to change anything else in the package

ViB

### CODE REVIEW

**Review by kmorel — DISMISSED** [review]
Looks good to me.

**Review by tldahlgren — COMMENTED** [review]
Confirmed sha256 for the new version.

Not sure if a rebase to develop will fix the `meson` issue or if that is internal.  There are 3 CI stacks that are failing at least the one I checked appears in relation to `vtk-m`.

**Comment by tldahlgren** [comment]
A rebase to `develop` should fix the package-audits checks failure since the PR that addressed that problem was merged about 9 hours ago.

**Comment by kwryankrattiger** [comment]
@vicentebolea Does this also need to have a constraint on the ROCm version that works with 1.8? I think I remember you saying there was a version that it needed, but I could be wrong.

**Comment by vicentebolea** [comment]
@kwryankrattiger yes, 4.3 and 4.5 works but with high memory usage and long building times 8 hours. 5.1 works the best with much lower memory usage and building times.

**Comment by kwryankrattiger** [comment]
Would it make sense to constrain the VTK-m build to `when='^hip@5.1:'` then? I feel like it may make sense to constrain this since it sounds like the older versions are not as useful/stable.

**Comment by vicentebolea** [comment]
@kwryankrattiger sure, all of the ecp machines I tries the have support for 5.1 already :sunglasses:

**Comment by vicentebolea** [comment]
I accidentally closed this while commenting :facepalm:

**Review by tldahlgren — DISMISSED** [review]
Re-confirmed sha256.

**Review by alalazo — CHANGES_REQUESTED** [review]
Can you check the build error in the pipelines?

**Comment by kwryankrattiger** [comment]
@vicentebolea It looks like vtk-m 1.8 has breaking API changes down stream. Can you also update those packages to require `@1.7:1.7`

**Comment by vicentebolea** [comment]
Contributors of edited packages in this pull-request. Does your package breaks with vtk-m 1.8, if not please let me know so that I can remove the change from your package.

File: var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ class EcpDataVisSdk(BundlePackage, CudaPackage, ROCmPackage):
                        when='+visit',
                        propagate=['hdf5', 'adios2'])
 
-    dav_sdk_depends_on('vtk-m@1.7:+shared+mpi+openmp+rendering',
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py]: Leave this I think, we want to be able to build a vtk-m 1.8 with ROCm if ascent is turned off.

**Review by kwryankrattiger — APPROVED** [review]

**Comment by tldahlgren** [comment]
@alalazo Looks like your review is blocking the merge of this PR

**Review by alalazo — APPROVED** [review]

**Pull request merged**